pthread_support.c

```diff
@@ -424,6 +425,16 @@ STATIC pthread_t GC_mark_threads[MAX_MARKERS];
         }
       }
 #   endif /* HPUX || GC_DGUX386_THREADS */
+
+    /* Apply special signal mask to GC marker threads, and don't drop user */
+    /* defined signals by GC marker threads.                               */
+    sigfillset(&set);
+    sigdelset(&set, GC_get_suspend_signal()    ); /* used by GC to stop the world    */
```

Comment:
Suspend/resume signals are not used for GC_DARWIN_THREADS, GC_OPENBSD_UTHREADS and NACL